sklearn/linear_model/tests/test_base.py

```diff
@@ -89,6 +91,16 @@ def test_linear_regression_sample_weights():
                 assert_almost_equal(inter1, coefs2[0])
 
 
+def test_raises_value_error_if_positive_and_sparse():
+    # X, y must not be sparse if positive == True
+    X = sparse.eye(10)
+    y = np.ones(10)
+
+    reg = LinearRegression(positive=True)
+
+    assert_raises(TypeError, reg.fit, X, y)
```

Comment:
Better to use,
```py
with pytest.raises(TypeError):
   reg.fit(X, y)
```

we are in the process of deprecating `assert_raises` utils function.